features/workflow-files.feature

```diff
@@ -0,0 +1,35 @@
+# This file is part of REANA.
+# Copyright (C) 2023 CERN.
+#
+# REANA is free software; you can redistribute it and/or modify it
+# under the terms of the MIT License; see LICENSE file for more details.
+
+# This is a behavioural test file for the `reana-demo-root6-roofit` repository.
+
+Feature: Files in the workspace
+
+    The workflow takes `gendata.C` and `fitdata.C` as input files,
+    runs them and produces `data.root` and `plot.png` as output files.
+
+    Scenario: The workspace contains the expected output files
+        When the workflow finishes
+        Then the status should be "finished"
+        And the workspace should include "results/data.root"
+        And the workspace should contain "/code/fitdata.C"
+        And all the outputs should be included in the workspace
+
+    Scenario: The workspace contains the expected input files
+        When the workflow finishes
+        Then the status should be "finished"
+        And the workspace should include "code/gendata.C"
+        And the workspace should include "code/fitdata.C"
+
+    Scenario: The code used to generate the plot is correct
+        When the workflow finishes
+        Then the file "code/fitdata.C" should contain "RooWorkspace* w = (RooWorkspace*) f->Get("w")"
+
+    Scenario: The size of the files in the workspace is not too large
```

Comment:
We can get to this one once we amend the log messages (see long comment above). E.g. ordering can be based more logically to first tests inputs, then outputs).  And precise file sizes are preferred over informal "not too large" phrase (that can be used only for workspace total size, but even there we'd better rename it more positively as "workspace total size remains moderate" or some such)